app/V1Module/presenters/InstancesPresenter.php

```diff
@@ -178,30 +184,11 @@ public function actionGroups(string $id) {
       throw new ForbiddenRequestException();
     }
 
-    $this->sendSuccessResponse(array_values(array_filter($instance->getGroups()->getValues(), function (Group $group) {
-      return $this->groupAcl->canViewDetail($group);
-    })));
-  }
-
-  /**
-   * Get a list of all public groups in an instance.
-   * @GET
-   * @param string $id An identifier of the instance
-   * @throws ForbiddenRequestException
-   */
-  public function actionPublicGroups(string $id) {
-    $instance = $this->instances->findOrThrow($id);
-    if (!$this->instanceAcl->canViewGroups($instance)) {
```

Comment:
Doesn't removing this mean that we can remove the `canViewGroups` privilege too?

Reply:
Not really sure... maybe... I'll look into it

Reply:
I see. Maybe stuff like [this](https://github.com/ReCodEx/api/blob/master/app/V1Module/presenters/InstancesPresenter.php#L179) should use GroupViewFactory too?

Reply:
You mean... like this: https://github.com/ReCodEx/api/blob/private-public-info/app/V1Module/presenters/InstancesPresenter.php#L191 ... ?

Reply:
Yeah, sorry for not checking the right branch.